src/util/TypeGuards.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const { ApplicationCommandOptionTypes } = require('./Constants');
+
+/**
+ * Helpers for checking different variants of common types.
+ */
+class TypeGuards extends null {
+  /**
+   * Resolves a given type to an enum equivalent value, and
+   * checks if it's part of the given enum type.
+   * @private
+   * @param {string|number} type The type to resolve
+   * @param {Object} object The enum to resolve to
+   * @param {string[]} fields The enum fields to check
+   * @returns {boolean} Whether the type is part of the enum
```

Comment:
We aren't super consistent with this, but I think putting private below returns is how we do it in most cases:
```suggestion
   * @param {string|number} type The type to resolve
   * @param {Object} object The enum to resolve to
   * @param {string[]} fields The enum fields to check
   * @returns {boolean} Whether the type is part of the enum
   * @private
```